scuttlebutt/src/server.rs

```diff
@@ -172,30 +173,113 @@ impl UdpServer {
 
     /// Gossip to multiple randomly chosen nodes.
     async fn gossip_multiple(&self, rng: &mut SmallRng) {
-        // TODO: Gossip with live nodes & randomly include dead node
+        // Gossip with live nodes & probabilistically include a random dead node
         let scuttlebutt_guard = self.scuttlebutt.lock().await;
-        let self_node_id = &scuttlebutt_guard.self_node_id;
+        let live_nodes_count = scuttlebutt_guard.live_nodes().count();
+        let dead_nodes_count = scuttlebutt_guard.dead_nodes().count();
         const EMPTY_STRING: String = String::new();
         let mut rand_nodes = [EMPTY_STRING; GOSSIP_COUNT];
 
-        // Select up to [`GOSSIP_COUNT`] node IDs at random.
-        let nodes = scuttlebutt_guard.cluster_state.nodes();
-        let count = nodes
-            .filter(|node_id| node_id != self_node_id)
+        // Select `GOSSIP_COUNT` number of live nodes.
+        // On startup, select from cluster nodes since we don't know any live node yet.
+        let count = if live_nodes_count == 0 {
```

Comment:
Would it be simpler to consider the seed nodes live on startup?

Reply:
The `cluster_nodes` will most likely contain seed nodes so I felt it's better. Besides this is just a case for the startup, in normal operation, we want live nodes.